omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/RiskFactorsFragmentController.java

```diff
@@ -0,0 +1,136 @@
+package org.openmrs.module.pihcore.fragment.controller.dashboardwidgets;
+
+import org.apache.commons.lang.StringUtils;
+import org.codehaus.jackson.JsonNode;
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.PatientProgram;
+import org.openmrs.Program;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.ProgramWorkflowService;
+import org.openmrs.module.appframework.domain.AppDescriptor;
+import org.openmrs.module.emrapi.patient.PatientDomainWrapper;
+import org.openmrs.ui.framework.annotation.FragmentParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.fragment.FragmentConfiguration;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class RiskFactorsFragmentController {
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    public void controller(@SpringBean("patientService") PatientService patientService,
+                           @SpringBean("conceptService") ConceptService conceptService,
+                           @SpringBean("obsService") ObsService obsService,
+                           @SpringBean("programWorkflowService") ProgramWorkflowService programWorkflowService,
+                           @FragmentParam("app") AppDescriptor app,
+                           FragmentConfiguration config,
+                           FragmentModel model) {
+
+        Object patientConfig = config.get("patient");
+        if (patientConfig == null) {
+            patientConfig = config.get("patientId");
+        }
+        Patient patient = null;
+        if (patientConfig != null) {
+            if (patientConfig instanceof Patient) {
+                patient = (Patient) patientConfig;
+            } else if (patientConfig instanceof PatientDomainWrapper) {
+                patient = ((PatientDomainWrapper) patientConfig).getPatient();
+            } else if (patientConfig instanceof Integer) {
+                patient = patientService.getPatient((Integer) patientConfig);
+            } else if (patientConfig instanceof String) {
+                patient = patientService.getPatientByUuid((String) patientConfig);
+            }
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
```

Comment:
It would be nice if we didn't have all of this mess supporting different ways of loading the patient.  I know this is in a lot of existing coreapps controllers, but this is brand new, and something only we are including.  Presumably we know exactly how the patient will be pass in all the time, and we can assume that?

Reply:
Yeah, agree it makes sense to simplify that.